### DISCUSSION TITLE
Clearing storage between user sessions

### DISCUSSION THREAD

**Discussion by zilch42** [zauberzeug/nicegui#2428] [Q&A] [answered]
### Question

If I wanted to use `app.storage.user` to store user data while a user was using the app, but not have it persist when they come back the next session (day, or whenever?), would this be the correct way to do it? Would this have any implications for other users currently using the app? I'm guessing not because startup would generally only happen when the first user came along, and this would clear all users who had previously used it, but there shouldn't be any other active users who might get their storage wiped mid session?

```python
def handle_startup():
    app.storage.clear()

app.on_startup(handle_startup)
```

I tried `app.storage.user.clear()` but that said `app.storage.user` can only be used within a page builder. 

If I used `on_connect` rather than `on_startup` is here a chance a user might reconnect during what they would consider 'a session'?
Actually, if I used `on_connect`, then I imagine that could clear the storage for other active users...?

**Comment by rodja** [comment, marked as answer]
Yes, you can to clear storage data on startup. But this is triggered directly after the startup of the server is done. Not when the first user connects. Normally you would not restart a server regularly (because that has the effect of disconnecting dropping all currently connected users). 

> If I used on_connect rather than on_startup is here a chance a user might reconnect during what they would consider 'a session'?
> Actually, if I used on_connect, then I imagine that could clear the storage for other active users...?

Exactly. It will get called for each connection which is not what you want.

Maybe the open feature request of [`app.storage.tab`](https://github.com/zauberzeug/nicegui/discussions/1308) is more what you really want? In https://github.com/zauberzeug/nicegui/discussions/2378 there is some code you could try...

**Reply by zilch42** [reply]
Thanks, I'll check that out!

**Comment by rodja** [comment]
We are currently working on two more storage types: [`app.storage.tab`](https://github.com/zauberzeug/nicegui/pull/2837) which provides a place to store data across reloads in a single tab and [`app.storage.client`](https://github.com/zauberzeug/nicegui/pull/2820) which holds data only for the length of the current client connection (eg. will drop on every page change).
